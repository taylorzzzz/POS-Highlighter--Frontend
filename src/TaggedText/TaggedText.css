@import '../colors.css';

.TaggedText {  
  padding: 40px 20px 20px;
  line-height: 2em;
  position: relative;
  letter-spacing: .05em;

  .adjective { 
    color: var(--color-adjective);
    &.underline {
      border-bottom-color: var(--color-adjective);
    }
  }

  .adverb { 
    color:  var(--color-adverb);
    &.underline {
      border-bottom-color: var(--color-adverb);
    }
  }

  .conjunction { 
    color: var(--color-conjunction);
    &.underline {
      border-bottom-color: var(--color-conjunction);
    }
  }

  .determiner { 
    color: var(--color-determiner);
    &.underline {
      border-bottom-color: var(--color-determiner);
    }
  }

  .noun { 
    color: var(--color-noun);
    &.underline {
      border-bottom-color: var(--color-noun);
    }
  }

  .number { 
    color: var(--color-number);
    &.underline {
      border-bottom-color: var(--color-number);
    }
  }

  .other { 
    color: var(--color-other);
    &.underline {
      border-bottom-color: var(--color-other);
    }
  }

  .preposition { 
    color: var(--color-preposition);
    &.underline {
      border-bottom-color: var(--color-preposition);
    }
  }

  .pronoun { 
    color: var(--color-pronoun);
    &.underline {
      border-bottom-color: var(--color-pronoun);
    }
  }

  .punctuation { 
    color: var(--color-punctuation);
    &.underline {
      border-bottom-color: var(--color-punctuation);
    }
  }

  .verb { 
    color: var(--color-verb);
    &.underline {
      border-bottom-color: var(--color-verb);
    }
  }

  .underline {
    border-bottom: 1px solid;
    color: inherit;
  }
  
}

.grapheme {
  display: inline;
}

.underline-toggle {
  position: absolute;
  top: 0px;
  left: 20px;
}
