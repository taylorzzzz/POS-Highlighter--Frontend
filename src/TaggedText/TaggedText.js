import React from 'react';

import './TaggedText.css';

function TaggedText({ markup, underline, toggleUnderline })  {

  console.log('rendering taggedtext');
  console.log(toggleUnderline);
  return (
    <div className="TaggedText">

      { markup }

      <div className="underline-toggle">
        <input 
          type="checkbox"
          checked={ underline }
          onChange={ toggleUnderline }
          id="underline">
        </input>
        <label htmlFor="underline">Underline</label>
      </div>

    </div>
  )

}

export default TaggedText;