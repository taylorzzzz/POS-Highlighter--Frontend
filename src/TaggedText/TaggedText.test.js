import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import TaggedText from './TaggedText';

test('TaggedText renders properly', () => {
    const tree = renderer.create(<TaggedText />).toJSON();
    
    expect(tree).toMatchSnapshot();
})

test('TaggedText toggleUnderline events properly', () => {
    const toggleSpy = jest.fn();

    const wrapper = shallow(<TaggedText toggleUnderline={toggleSpy} />);
    
    expect(toggleSpy).toHaveBeenCalledTimes(0);

    wrapper.find('input').simulate('change');
    
    expect(toggleSpy).toHaveBeenCalled();
})

test('TaggedText underline prop correctly sets checked value to false', () => {
    
    const wrapper = shallow(<TaggedText underline={false}/>);
    
    const input = wrapper.find('input');

    expect(input.props().checked).toBeFalsy();
})

test('TaggedText underline prop correctly sets checked value to true', () => {
    
    const wrapper = shallow(<TaggedText underline={true}/>);
    
    const input = wrapper.find('input');

    expect(input.props().checked).toBeTruthy();
})

