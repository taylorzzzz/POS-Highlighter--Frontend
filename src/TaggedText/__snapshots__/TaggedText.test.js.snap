// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaggedText renders properly 1`] = `
<div
  className="TaggedText"
>
  <div
    className="underline-toggle dot"
  >
    <input
      checked={undefined}
      id="underline"
      onChange={undefined}
      type="checkbox"
    />
    <label
      htmlFor="underline"
    >
      Underline
    </label>
  </div>
</div>
`;
